plan/point_get_plan.go

```diff
@@ -193,15 +193,19 @@ func tryPointGetPlan(ctx sessionctx.Context, selStmt *ast.SelectStmt) *PointGetP
 			return nil
 		}
 		p := newPointGetPlan(ctx, schema, tbl)
-		p.Handle = handleDatum.GetInt64()
+		var err error
+		p.Handle, err = handleDatum.ToInt64(ctx.GetSessionVars().StmtCtx)
+		if err != nil {
+			return nil
+		}
 		return p
 	}
 	for _, idxInfo := range tbl.Indices {
 		if !idxInfo.Unique {
 			continue
 		}
 		if idxInfo.State != model.StatePublic {
-			return nil
+			continue
```

Comment:
We should skip the non-public indexes to check all the public indexes, so I changed the `return` instruction to `continue`